### PR TITLE
Queue | Adds endpoint to get tasks by VACOLS ID

### PR DESCRIPTION
**Pull request by evankroske**
Part of https://github.com/department-of-veterans-affairs/caseflow/issues/6222

### Description
Please explain the changes you made here.

### Testing Plan
- [ ] Become BVAAABSHIRE.
- [ ] View http://localhost:3000/appeals/3449656/tasks .
- [ ] Confirm that the endpoint returned a JSON task.
- [ ] Repeat as BVARERDMAN.
- [ ] Repeat as BVALSPORER.
- [ ] Repeat as BVASCASPER1.

### CODE REVIEW

File: app/controllers/appeals_controller.rb [hunk 1]
```diff
@@ -21,6 +21,15 @@ def document_count
     render e.serialize_response
   end
 
+  def tasks
+    no_cache
+
+    tasks, _ = LegacyWorkQueue.tasks_with_appeals_of_appeal(params[:appeal_id])
```
Comment (aroltsch) [inline app/controllers/appeals_controller.rb]: Are you sending vacols ID or appeal ID? I think we should be sending appeal ID and then getting vacols id from the appeal object

File: app/controllers/legacy_tasks_controller.rb [hunk 2]
```diff
@@ -16,6 +16,17 @@ def index
         render "queue/show"
       end
       format.json do
+        if params[:appeal_id]
```
Comment (aroltsch) [inline app/controllers/legacy_tasks_controller.rb]: Why did you add it here as well?

File: app/models/queues/legacy_work_queue.rb [hunk 3]
```diff
@@ -7,17 +7,27 @@ class << self
...
+    end
 
-      tasks = vacols_tasks.zip(vacols_appeals).map do |task, appeal|
-        (role.capitalize + "LegacyTask").constantize.from_vacols(task, appeal, user)
-      end
-      [tasks, vacols_appeals]
+    def tasks_with_appeals_of_appeal(appeal_id)
+      vacols_tasks = repository.tasks_for_appeal(appeal_id)
+      # Look up user and their role
+      tasks_with_appeals_of_vacols_tasks(user, role, vacols_tasks)
```
Comment (aroltsch) [inline app/models/queues/legacy_work_queue.rb]: You left a comment here probably forgot to implement. However, do we even need a user for this? Probably not. We need a role though so we know which class to instantiate.

File: app/controllers/appeals_controller.rb (hunk 1 above)
Comment (aroltsch) [reply app/controllers/appeals_controller.rb]: If we don't need appeals, should we change the method to return tasks only? I would name it `tasks_by_appeal_id`

**Comment by evankroske** [comment]
I decided to fix this issue another way: https://github.com/department-of-veterans-affairs/caseflow/pull/6245

File: app/controllers/appeals_controller.rb (hunk 1 above)
Comment (evankroske) [reply app/controllers/appeals_controller.rb]: I'm sending a vacols ID because that's what /queue/appeals/X uses.

I made the method return tasks and appeals even though we don't need appeals because we have to create the appeal objects in order to get the tasks. Since we're creating the objects, I thought that we might as well return them. However, I could be convinced that we should change tasks_with_appeals_of_appeal to only return tasks. What do you think?

File: app/controllers/legacy_tasks_controller.rb (hunk 2 above)
Comment (evankroske) [reply app/controllers/legacy_tasks_controller.rb]: Sorry, I forgot to remove this code. I'll get rid of it

Comment (evankroske) [reply app/controllers/legacy_tasks_controller.rb]: Done!

File: app/controllers/legacy_tasks_controller.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@ class LegacyTasksController < ApplicationController
   before_action :verify_queue_access
   before_action :verify_task_assignment_access, only: [:create, :update]
 
-  ROLES = %w[judge attorney].freeze
+  ROLES = %w[judge attorney colocated].freeze
```
Comment (aroltsch) [inline app/controllers/legacy_tasks_controller.rb]: Why did you add colocated here?

File: app/models/queues/legacy_work_queue.rb [hunk 5]
```diff
@@ -7,17 +7,30 @@ class << self
 
     def tasks_with_appeals(user, role)
       vacols_tasks = repository.tasks_for_user(user.css_id)
-      vacols_appeals = repository.appeals_from_tasks(vacols_tasks)
+      tasks_with_appeals_of_vacols_tasks(user, role, vacols_tasks)
+    end
 
-      tasks = vacols_tasks.zip(vacols_appeals).map do |task, appeal|
-        (role.capitalize + "LegacyTask").constantize.from_vacols(task, appeal, user)
+    def tasks_with_appeals_of_appeal(appeal_id)
```
Comment (aroltsch) [inline app/models/queues/legacy_work_queue.rb]: It might be me but I think "of" appeal is not what we usually use to name methods. I think "by" is more clear. However, I will leave it up to you.

File: app/models/queues/legacy_work_queue.rb [hunk 6]
```diff
@@ -7,17 +7,30 @@ class << self
...
+      user = User.find_or_create_by(css_id: vacols_tasks[0].staff.sdomainid, station_id: User::BOARD_STATION_ID)
+      tasks_with_appeals_of_vacols_tasks(user, user.vacols_roles.first, vacols_tasks)
     end
 
     def repository
       return QueueRepository if FeatureToggle.enabled?(:test_facols)
       @repository ||= QueueRepository
     end
+  private
+    def tasks_with_appeals_of_vacols_tasks(user, role, vacols_tasks)
```
Comment (aroltsch) [inline app/models/queues/legacy_work_queue.rb]: I would personally rename 
`tasks_with_appeals` => `tasks_with_appeals_by_user_id`
`tasks_with_appeals_of_appeal` => `tasks_with_appeals_by_appeal_id`
`tasks_with_appeals_of_vacols_tasks` => `tasks_with_appeals`

File: app/models/queues/legacy_work_queue.rb [hunk 7]
```diff
@@ -7,17 +7,30 @@ class << self
...
 
-      tasks = vacols_tasks.zip(vacols_appeals).map do |task, appeal|
-        (role.capitalize + "LegacyTask").constantize.from_vacols(task, appeal, user)
+    def tasks_with_appeals_of_appeal(appeal_id)
+      vacols_tasks = repository.tasks_for_appeal(appeal_id)
+      if vacols_tasks.empty?
+        return [], []
       end
-      [tasks, vacols_appeals]
+      user = User.find_or_create_by(css_id: vacols_tasks[0].staff.sdomainid, station_id: User::BOARD_STATION_ID)
```
Comment (aroltsch) [inline app/models/queues/legacy_work_queue.rb]: Hmmm. Not sure if this is 100% correct. So any user can query any task and then we assume this returned user associated with the task works at the board.

File: app/models/queues/legacy_work_queue.rb [hunk 8]
```diff
@@ -7,17 +7,30 @@ class << self
...
+      tasks_with_appeals_of_vacols_tasks(user, user.vacols_roles.first, vacols_tasks)
```
Comment (aroltsch) [inline app/models/queues/legacy_work_queue.rb]: User can have multiple roles, so we need to pass `role` as a parameter to specify which tasks to return.

File: app/models/queues/legacy_work_queue.rb (hunk 7 above)
Comment (aroltsch) [reply app/models/queues/legacy_work_queue.rb]: Thinking about this code more. I can see the complexity. Let's chat about this in person.

Comment (aroltsch) [reply app/models/queues/legacy_work_queue.rb]: I think if you planning on reusing moss of the code from here we should just move the appeal_id as an optional parameter to GET /legacy_tasks?user_id=xxx&role=xxxx. Then if the `appeal_id` is present, filter out tasks all the way at the end to only return tasks associated with the appeal.

File: app/models/queues/legacy_work_queue.rb (hunk 6 above)
Comment (evankroske) [reply app/models/queues/legacy_work_queue.rb]: I renamed tasks_with_appeals_of_appeal, but I'd prefer not to rename the rest because tasks_with_appeals is used a few places.

File: app/models/queues/legacy_work_queue.rb (hunk 7 above)
Comment (evankroske) [reply app/models/queues/legacy_work_queue.rb]: I changed the endpoint to require a role parameter
